shuriken/include/shuriken/parser/Macho/macho_header.h

```diff
@@ -0,0 +1,53 @@
+//--------------------------------------------------------------------*- C++ -*-
+// Shuriken-Analyzer: library for bytecode analysis.
+// @author Farenain <kunai.static.analysis@gmail.com>
+//
+// @file header.h
+// @brief MachoHeader of a MACHO file represented by a structure
+
+#ifndef SHURIKENLIB_MACHO_HEADER_H
+#define SHURIKENLIB_MACHO_HEADER_H
+
+#include "shuriken/common/shurikenstream.h"
+
+#include "cstring"
+
+namespace shuriken::parser::macho {
+    class MachoHeader {
+    public:
+#pragma pack(1)
```

Comment:
can i ask why we're using pack here?

Reply:
This is correct to avoid some issue in case some padding is included in the structure. When reading from a data stream and you read a full structure, you can have a byte mismatch if you have some padding.